src/main/java/hudson/plugins/s3/utils/S3Utils.java

```diff
@@ -0,0 +1,64 @@
+package hudson.plugins.s3.utils;
+
+import hudson.ProxyConfiguration;
+import hudson.util.Secret;
+
+import java.util.regex.Pattern;
+
+import jenkins.model.Jenkins;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider;
+import com.amazonaws.internal.StaticCredentialsProvider;
+import com.amazonaws.services.s3.AmazonS3Client;
+
+public class S3Utils {
+
+    public static AmazonS3Client createClient(String accessKey, Secret secretKey, boolean useRole, String stsRoleArn) {
+        AWSCredentialsProvider credentialsProvider;
+        if (useRole) {
+            credentialsProvider = new DefaultAWSCredentialsProviderChain();
```

Comment:
This may lead to a regression as it will prevent anonymous access.

The [`S3CredentialsProviderChain`](https://github.com/aws/aws-sdk-java/blob/6caeef80c08583d3f70d3bcf40ca7715edb906de/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/AmazonS3Client.java#L409) provided by the constructor [`AmazonS3Client(ClientConfiguration clientConfiguration)`](https://github.com/aws/aws-sdk-java/blob/6caeef80c08583d3f70d3bcf40ca7715edb906de/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/AmazonS3Client.java#L613) is unfortunately slightly different from  the `DefaultAWSCredentialsProviderChain` as the `S3CredentialsProviderChain` defaults to anonymous access. 

To make it more difficult, the `S3CredentialsProviderChain` is a private class we can't use.

https://github.com/aws/aws-sdk-java/blob/6caeef80c08583d3f70d3bcf40ca7715edb906de/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/AmazonS3Client.java#L409